src/SelfServe/Documentation/README.md

```diff
@@ -0,0 +1,296 @@
+- No need to ramp up on the intricacies of every feature which requires UI changes
+- Own only the framework and not every feature, giving more bandwidth to prioritize inhouse features as well
+
+## Getting Started
+
+Clone the cosmos-explorer repo and run
+
+- `npm install`
+- `npm run build`
+
+## Code Changes
+
+Code changes need to be made only in the following files
+- Strings JSON file
+- Types File
+- RP file
+- Class file
+- SelfServeType in [SelfServeUtils.tsx](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/SelfServeUtils.tsx)
+- [SelfServe.tsx](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/SelfServe.tsx)
+
+### 1. JSON file for UI strings
+
+#### Naming Convention
+`Localization/en/<FEATURE_NAME>.json`\
+Please place your files only under "Localization/en" folder. If not, the UI strings will not be picked up by the framework.
+
+#### Example
+[SelfServeExample.json](https://github.com/Azure/cosmos-explorer/blob/master/src/Localization/en/SelfServeExample.json)
+
+#### Description
+This is a JSON file where the values are the strings that needs to be displayed in the UI. These strings are referenced using their corresponding unique keys.
+
+For example, If your class file defines properties as follows
+```ts
+  @Values({
+    labelTKey: "stringPropertylabel"
+  })
+  stringProperty: string;
+
+  @Values({
+    labelTKey: "booleanPropertyLabel",
+    trueLabelTKey: "trueLabel",
+    falseLabelTKey: "falseLabel",
+  })
+  booleanProperty: boolean;
+```
+
+Then the content of `Localization/en/FeatureName.json` should be 
+
+```json
+{
+    stringPropertyLabel: "string property",
+    booleanPropertyLabel: "boolean property",
+    trueLabel: "Enable",
+    falseLabel: "Disable"
+}
+```
+You can learn more on how to define the class file [here](./selfserve.html#4-class-file).
+
+### 2. Types file
+
+#### Naming Convention
+`<FEATURE_NAME>.types.ts`
+
+#### Example
+[SelfServeExample.types.ts](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/Example/SelfServeExample.types.ts)
+
+#### Description
+This file contains the definitions of all the types and interfaces to be used in your Class file and RP file.
+
+For example, if you want to save the `stringProperty` and `booleanProperty` of your SelfServe class, then you can define an interface in your `FeatureName.types.ts` file like this.
+
+```ts
+export PropertiesToSave {
+  stringProperty: string,
+  booleanProperty: boolean
+}
+```
+
+### 3. RP file
+
+#### Naming Convention
+`<FEATURE_NAME>.rp.ts`
+
+#### Example
+[SelfServeExample.rp.ts](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/Example/SelfServeExample.rp.ts)
+
+#### Description
+The RP file will host the REST calls needed for the initialize, save and refresh functions. This decouples the code that builds the UI and the code that makes REST calls.
+
+The `userContext` object exposes the following Azure Cosmos DB databaseAccount information
```

Comment:
>The [](http://example.com/codeflow?start=0&length=3)

some phrase for sentence and context continuity here: 
eg: Before we make API call, we need som basic info like accname, sub,, which is readily available from the userContext object.